Xamarin.Forms.Platform.Android/FastRenderers/AutomationPropertiesProvider.cs

```diff
@@ -46,17 +46,23 @@ internal static void SetAutomationId(AView control, VisualElement element, strin
 
 		internal static void SetBasicContentDescription(
 			AView control,
-			BindableObject element,
+			BindableObject bindableObject,
 			ref string defaultContentDescription)
 		{
-			if (element == null || control == null)
+			if (bindableObject == null || control == null)
 				return;
 
 			if (defaultContentDescription == null)
 				defaultContentDescription = control.ContentDescription;
 
-			string value = ConcatenateNameAndHelpText(element);
-			control.ContentDescription = !string.IsNullOrWhiteSpace(value) ? value : defaultContentDescription;
+			string value = ConcatenateNameAndHelpText(bindableObject);
+
+			var contentDescription = !string.IsNullOrWhiteSpace(value) ? value : defaultContentDescription;
+
+			if (String.IsNullOrWhiteSpace(contentDescription) && bindableObject is Element element)
+				contentDescription = element.AutomationId;
```

Comment:
@samhouts is this ok? It doesn't seem like the *ContentDescription* should get overwritten here if the user has set the AutomationId to something

Reply:
works! ship it